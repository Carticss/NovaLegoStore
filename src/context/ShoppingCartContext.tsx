import React, { createContext, useState, useEffect } from 'react';
import AsyncStorage from '@react-native-community/async-storage';

const STORAGE_KEY = '@ShoppingCart:items';

interface ShoppingCartItem {
  id: number;
  name: string;
  unit_price: number;
  stock: number;
  image: string;
}

interface ShoppingCartContextType {
  items: ShoppingCartItem[];
  addItem: (item: ShoppingCartItem) => void;
  removeItem: (item: ShoppingCartItem) => void;
  payAll: () => void;
}

export const ShoppingCartContext = createContext<ShoppingCartContextType>({
  items: [],
  addItem: (item: ShoppingCartItem) => {},
  removeItem: (item: ShoppingCartItem) => {},
  payAll: () => {},
});

export const ShoppingCartProvider = (props: { children: React.ReactNode }) => {
    
  const [items, setItems] = useState<ShoppingCartItem[]>([]);

  const saveItems = async () => {
    try {
      await AsyncStorage.setItem(STORAGE_KEY, JSON.stringify(items));
    } catch (e) {
      console.error(e);
    }
  };

  const loadItems = async () => {
    try {
      const storedItems = await AsyncStorage.getItem(STORAGE_KEY);
      if (storedItems) {
        setItems(JSON.parse(storedItems));
      }
    } catch (e) {
      console.error(e);
    }
  };

  useEffect(() => {
    loadItems();
  }, []);

  useEffect(() => {
    saveItems();
  }, [items]);

  const addItem = (item: ShoppingCartItem) => {
    setItems(prevItems => [...prevItems, item]);
  };

  const removeItem = async (item: ShoppingCartItem) => {
    const newItems = items
    newItems.splice(newItems.indexOf(item), 1)
    await AsyncStorage.setItem(STORAGE_KEY, JSON.stringify(newItems));
    loadItems();
    setItems(newItems);
  };

  const payAll = async () => {
    await AsyncStorage.removeItem(STORAGE_KEY);
    loadItems();
    setItems([]);
  };

  return (
    <ShoppingCartContext.Provider
      value={{
        items,
        addItem,
        removeItem,
        payAll
      }}
    >
      {props.children}
    </ShoppingCartContext.Provider>
  );
};
