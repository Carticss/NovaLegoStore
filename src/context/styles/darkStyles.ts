import { StyleSheet, Dimensions } from "react-native";

const { width, height } = Dimensions.get("window");

export const darkStyles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: "#212121",
        justifyContent: "center",
        alignItems: "center",
        width: width,
    },
    formContainer: {
        justifyContent: "center",
        alignItems: "center",
        width: "100%",
        height: "100%",
    },
    input: {
        width: "95%",
        height: width * 0.13,
        borderRadius: 20,
        backgroundColor: "#4b4b4b",
        fontSize: 15,
        color: "white",
        paddingStart: "5%",
        fontWeight: "bold",
        fontFamily: "URW-Geometric-Regular",
    },
    legoIcon: {
        width: width * 0.2,
        height: width * 0.2,
        alignSelf: "flex-start",
    },
    bagIcon: {
        width: width * 0.06,
        height: width * 0.06,
    },
    arrowIcon: {
        width: width * 0.1,
        height: width * 0.065,
    },
    title: {
        fontSize: 30,
        fontWeight: "bold",
        color: "white",
        textAlign: "center",
        fontFamily: "URW-Geometric-Regular",
        paddingStart: 10,
    },
    logoContainer: {
        display: "flex",
        flexDirection: "row",
        justifyContent: "center",
        alignItems: "center",
        width: "100%",
        position: "absolute",
        top: height * 0.1,
    },
    arrowContainer: {
        display: "flex",
        flexDirection: "row",
        justifyContent: "center",
        alignItems: "center",
        width: "20%",
        position: "absolute",
        top: "3.5%",
        overflow: "visible",
        alignSelf: "flex-start",
    },
    label: {
        color: "white",
        fontSize: 15,
        fontWeight: "bold",
        fontFamily: "URW-Geometric-Regular",
        alignSelf: "flex-start",
        marginStart: "5%",
        marginBottom: 5,
    },
    loginButton: {
        width: "65%",
        height: width * 0.13,
        borderRadius: 20,
        backgroundColor: "#0073ff",
        justifyContent: "center",
        alignItems: "center",
        marginTop: 30,
    },
    loginButtonText: {
        paddingHorizontal: "20%",
        color: "white",
        fontSize: 15,
        fontWeight: "bold",
        fontFamily: "URW-Geometric-Regular",
    },
    switch: {
        width: "20%",
        height: width * 0.08,
        borderRadius: 20,
        backgroundColor: "#0073ff",
        justifyContent: "center",
        alignItems: "center",
        position: "absolute",
        right: "5%",
        top: "3%",
    },
    switchBottom: {
        width: "20%",
        height: width * 0.08,
        borderRadius: 20,
        backgroundColor: "#0073ff",
        justifyContent: "center",
        alignItems: "center",
        position: "absolute",
        right: "5%",
        top: "91%",
    },
    logOutBottom: {
        width: "20%",
        height: width * 0.08,
        borderRadius: 20,
        backgroundColor: "#0073ff",
        justifyContent: "center",
        alignItems: "center",
        position: "absolute",
        right: "5%",
        top: "84%",
    },
    payAllBottom: {
        width: "20%",
        height: width * 0.08,
        borderRadius: 20,
        backgroundColor: "#0073ff",
        justifyContent: "center",
        alignItems: "center",
        position: "absolute",
        right: "5%",
        top: "77%",
    },
    switchPlchldr: {
        paddingHorizontal: "10%",
        color: "black",
        fontSize: 15,
        fontWeight: "bold",
        fontFamily: "URW-Geometric-Regular",
    },
    whiteDot: {
        width: "35%",
        height: "80%",
        borderRadius: 50,
        backgroundColor: "white",
        position: "absolute",
        right: "9%",
    },
    errorMessage: {
        color: "#0073ff",
        fontSize: 15,
        fontWeight: "bold",
        fontFamily: "URW-Geometric-Regular",
    },
    footer: {
        width: "100%",
        position: "absolute",
        bottom: "10%",
        justifyContent: "center",
        alignItems: "center",
    },
    footerText: {
        color: "white",
        fontSize: 15,
        fontWeight: "bold",
        fontFamily: "URW-Geometric-Regular",
    },
    footerLink: {
        color: "#0073ff",
        fontSize: 15,
        fontWeight: "bold",
        fontFamily: "URW-Geometric-Regular",
    },
    bagContainer: {
        marginTop: "5%",
        marginRight: "5%",
        display: "flex",
        flexDirection: "row",
        justifyContent: "flex-end",
        alignItems: "center",
        width: "20%",
        height: "80%",
        overflow: "visible",
        backgroundColor: "#0073ff",
        borderRadius: 20,
    },
    productContainer: {
        flexDirection: 'row',
        alignItems: 'center',
        padding: 25,
        width: '100%',
    },
    productContainer2: {
        flexDirection: 'row',
        alignItems: 'center',
        padding: 25,
        width: '100%',
    },
    productImage: {
        width: 100,
        height: 100,
        resizeMode: 'contain',
        backgroundColor: 'white',
        borderRadius: 20,
        borderColor: '#0073ff',
        borderWidth: 3,
    },
    productDetailsContainer: {
        marginLeft: 16,
    },
    productName: {
        fontSize: 16,
        fontWeight: 'bold',
        color: 'white',
    },
    productPrice: {
        fontSize: 14,
        color: 'grey',
    },
    productStock: {
        fontSize: 14,
        color: 'lightgrey',
    },
    addToCartButton: {
        backgroundColor: "#0073ff",
        width: 100,
        height: 30,
        alignItems: 'center',
        justifyContent: 'center',
        marginTop: 10,
        borderRadius: 5
    },
    image: {
        width: 200,
        height: 200,
        resizeMode: 'contain',
        marginBottom: 16,
    },
    name: {
        fontSize: 24,
        fontWeight: 'bold',
        color: 'white',
    },
    price: {
        fontSize: 20,
        color: 'grey',
    },
    stock: {
        fontSize: 16,
        color: 'lightgrey',
    },
    description: {
        fontSize: 16,
        marginTop: 16,
        padding: 16,
        textAlign: 'center',
        color: 'lightgrey',
    },
});
