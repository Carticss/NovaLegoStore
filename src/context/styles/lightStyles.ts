import { StyleSheet, Dimensions } from "react-native";
import { darkStyles } from "./darkStyles";

const { width, height } = Dimensions.get("window");

export const lightStyles : typeof darkStyles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: "white",
        justifyContent: "center",
        alignItems: "center",
        width: width,
    },
    formContainer: {
        justifyContent: "center",
        alignItems: "center",
        width: "100%",
        height: "100%",
    },
    input: {
        width: "95%",
        height: width * 0.13,
        borderRadius: 20,
        backgroundColor: "#e6e6e6",
        fontSize: 15,
        color: "black",
        paddingStart: "5%",
        fontWeight: "bold",
        fontFamily: "URW-Geometric-Regular",
    },
    legoIcon: {
        width: width * 0.2,
        height: width * 0.2,
        alignSelf: "flex-start",
    },
    title: {
        fontSize: 30,
        fontWeight: "bold",
        color: "black",
        textAlign: "center",
        fontFamily: "URW-Geometric-Regular",
        paddingStart: 10,
    },
    logoContainer: {
        display: "flex",
        flexDirection: "row",
        justifyContent: "center",
        alignItems: "center",
        width: "100%",
        position: "absolute",
        top: height * 0.1,
    },
    label: {
        color: "grey",
        fontSize: 15,
        fontWeight: "bold",
        fontFamily: "URW-Geometric-Regular",
        alignSelf: "flex-start",
        marginStart: "5%",
        marginBottom: 5,
    },
    loginButton: {
        width: "65%",
        height: width * 0.13,
        borderRadius: 20,
        backgroundColor: "#ff8c00",
        justifyContent: "center",
        alignItems: "center",
        marginTop: 30,
    },
    loginButtonText: {
        paddingHorizontal: "20%",
        color: "white",
        fontSize: 15,
        fontWeight: "bold",
        fontFamily: "URW-Geometric-Regular",
    },
    switch: {
        width: "20%",
        height: width * 0.08,
        borderRadius: 20,
        backgroundColor: "#ff8c00",
        justifyContent: "center",
        alignItems: "center",
        position: "absolute",
        right: "5%",
        top: "3%",
    },
    switchPlchldr: {
        paddingHorizontal: "10%",
        color: "white",
        fontSize: 15,
        fontWeight: "bold",
        fontFamily: "URW-Geometric-Regular",
    },
    whiteDot: {
        width: "35%",
        height: "80%",
        borderRadius: 50,
        backgroundColor: "white",
        position: "absolute",
        right: "55%",
    },
    errorMessage: {
        color: "#ff8c00",
        fontSize: 15,
        fontWeight: "bold",
        fontFamily: "URW-Geometric-Regular",
    },
});
