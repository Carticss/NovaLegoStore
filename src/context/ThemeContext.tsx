import { createContext, Dispatch  } from "react";
import { darkStyles } from "./styles/darkStyles";
import { lightStyles } from "./styles/lightStyles";

export enum ThemeTypes {
    DARK = "dark",
    LIGHT = "light",
}

export interface ThemeAction {
    payload: any;
    type: ThemeTypes;
}

export interface ThemeContextProps {
    state: typeof darkStyles;
    dispatch: Dispatch<ThemeAction>;
}

export const ThemeContext = createContext<ThemeContextProps | null>(null);

export const ThemeActions = (state: typeof darkStyles, action: ThemeAction) => {
    switch (action.type) {
        case ThemeTypes.DARK:
            return darkStyles;
        case ThemeTypes.LIGHT:
            return lightStyles;
        default:
            return lightStyles;
    }
}