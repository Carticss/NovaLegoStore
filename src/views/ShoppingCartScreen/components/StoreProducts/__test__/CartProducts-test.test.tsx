/**
 * @format
 */

import 'react-native';
import React from 'react';
import CartProductList from '../CartProducts';
import { ThemeContext } from '../../../../../context/ThemeContext';
import { lightStyles } from '../../../../../context/styles/lightStyles';
import { render, screen, fireEvent } from '@testing-library/react-native';
import AsyncStorage from '@react-native-community/async-storage';

const products = [
  {
    "id": 1,
    "name": "Product 1",
    "unit_price": 100,
    "stock": 10,
    "image": "https://picsum.photos/200/300"
  },
  {
    "id": 1,
    "name": "Product 1",
    "unit_price": 100,
    "stock": 0,
    "image": "https://picsum.photos/200/300"
  },
]

beforeAll(() => {
  jest.mock('@react-native-community/async-storage');
})

describe('Tests in App.tsx', () => {
  test('should render correctly App.tsx', () => {
    const dispatch = jest.fn();
    const screen = render(
      <>
        <ThemeContext.Provider
          value={
            {
              state: lightStyles,
              dispatch: dispatch
            }
          }
        >
          <CartProductList products={products} />
        </ThemeContext.Provider>
      </>).toJSON();
    expect(screen).toMatchSnapshot();
  });

  test('should remove product to cart', () => { 
    const dispatch = jest.fn();
    const setItem = jest.fn(async () => {});
    jest.spyOn(AsyncStorage, 'setItem').mockResolvedValueOnce(new setItem);
    render(
      <>
        <ThemeContext.Provider
          value={
            {
              state: lightStyles,
              dispatch: dispatch
            }
          }
        >
          <CartProductList products={products} />
        </ThemeContext.Provider>
      </>).toJSON();
     fireEvent.press(screen.getAllByText('Remove')[0]);
      expect(setItem).toHaveBeenCalled();
  })
});
