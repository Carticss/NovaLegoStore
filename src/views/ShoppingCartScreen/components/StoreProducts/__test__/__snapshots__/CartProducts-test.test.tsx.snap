// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in App.tsx should render correctly App.tsx 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "padding": 25,
          "width": "100%",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://picsum.photos/200/300",
          }
        }
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#ff8c00",
            "borderRadius": 20,
            "borderWidth": 3,
            "height": 100,
            "resizeMode": "contain",
            "width": 100,
          }
        }
      />
      <View
        style={
          Object {
            "marginLeft": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Product 1
        </Text>
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 14,
            }
          }
        >
          $
          100
        </Text>
        <Text
          style={
            Object {
              "color": "lightgrey",
              "fontSize": 14,
            }
          }
        >
          10
           in stock
        </Text>
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ff8c00",
              "borderRadius": 5,
              "height": 30,
              "justifyContent": "center",
              "marginTop": 10,
              "opacity": 1,
              "width": 100,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "URW-Geometric-Regular",
                "fontSize": 15,
                "fontWeight": "bold",
                "paddingHorizontal": "10%",
              }
            }
          >
            Remove
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "padding": 25,
          "width": "100%",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://picsum.photos/200/300",
          }
        }
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#ff8c00",
            "borderRadius": 20,
            "borderWidth": 3,
            "height": 100,
            "resizeMode": "contain",
            "width": 100,
          }
        }
      />
      <View
        style={
          Object {
            "marginLeft": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Product 1
        </Text>
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 14,
            }
          }
        >
          $
          100
        </Text>
        <Text
          style={
            Object {
              "color": "lightgrey",
              "fontSize": 14,
            }
          }
        >
          0
           in stock
        </Text>
        <View
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "grey",
              "borderRadius": 5,
              "height": 30,
              "justifyContent": "center",
              "marginTop": 10,
              "opacity": 1,
              "width": 100,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "URW-Geometric-Regular",
                "fontSize": 15,
                "fontWeight": "bold",
                "paddingHorizontal": "10%",
              }
            }
          >
            Remove
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
